boilerplate/openshift/golang-osd-operator/csv-generate/csv-generate.sh

```diff
@@ -0,0 +1,127 @@
+        exit 1 
+    fi
+else
+    rm -rf "$SAAS_OPERATOR_DIR"
+    git clone --branch "$operator_channel" ${GIT_PATH} "$SAAS_OPERATOR_DIR"
+    
+    # remove any versions more recent than deployed hash
+    if [[ "$operator_channel" == "production" ]]; then
+        if [ -z "$DEPLOYED_HASH" ] ; then
+            DEPLOYED_HASH=$(
+                curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml" | \
+                    docker run --rm -i quay.io/app-sre/yq yq r - "resourceTemplates[*].targets(namespace.\$ref==/services/osd-operators/namespaces/hivep01ue1/${_OPERATOR_NAME}.yml).ref"
+            )
+        fi
+    
+        delete=false
+        # Sort based on commit number
+        for version in $(ls $BUNDLE_DIR | sort -t . -k 3 -g); do
+            # skip if not directory
+            [ -d "$BUNDLE_DIR/$version" ] || continue
+    
+            if [[ "$delete" == false ]]; then
+                short_hash=$(echo "$version" | cut -d- -f2)
+    
+                if [[ "$DEPLOYED_HASH" == "${short_hash}"* ]]; then
+                    delete=true
+                fi
+            else
+                rm -rf "${BUNDLE_DIR:?BUNDLE_DIR var not set}/$version"
+            fi
+        done
+    fi
+    # TODO : Clean handling of major version (should be variable from consumer repo and default to 0.1 is undefined)
+    OPERATOR_PREV_VERSION=$(ls "$BUNDLE_DIR" | sort -t . -k 3 -g | tail -n 1)
+    OPERATOR_NEW_VERSION="0.1.${operator_commit_number}-${operator_commit_hash}"
+    OUTPUT_DIR=${BUNDLE_DIR}
+fi
+
+if [[ "$generate_script" = "common" ]] ; then
+    ./boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py -o ${operator_name} -d ${OUTPUT_DIR} -p ${OPERATOR_PREV_VERSION} -n ${operator_commit_number} -c ${operator_commit_hash} -i ${operator_image}
+elif [[ "$generate_script" = "hack" ]] ; then
+    if [ -z "$OPERATOR_PREV_VERSION" ] ; then 
+        OPERATOR_PREV_VERSION="no-version"
+        DELETE_REPLACE=true
+    fi
+    
+    ./hack/generate-operator-bundle.py ${OUTPUT_DIR} ${OPERATOR_PREV_VERSION} ${operator_commit_number} ${operator_commit_hash} ${operator_image}
+    
+    if [ ! -z "${DELETE_REPLACE}" ] ; then
+        yq d -i output-comparison/${OPERATOR_NEW_VERSION}/*.clusterserviceversion.yaml 'spec.replaces'
+    fi
+fi
+
+if [ "$diff_generate" = true ] ; then
+    # TODO : Current hack script does not allow to generate the CSV for the comparison (it will generate a different version that the common one because there is 1 extra version in the history)
+    if [[ "$generate_script" = "hack" ]] ; then
+        echo "Generating with the common script and after, generating with the hack script is not supported yet. For comparison, please first generate with hack script, and then build/compare with the common script"
+        exit 1
```

Comment:
If we get here, we've already generated the bundle, which leaves a mess. This seems like it would be an easy condition to check early, before we've written anything to disk.

```bash
if [[ "$diff_generate" == "true" ]] && [[ "$generate_script" == "hack" ]]; then
    err "Generating .... is not supported ..."
fi
```